### PR TITLE
Support ext-session-lock protocol

### PR DESCRIPTION
**Pull request by johanmalm**
…because the next version of swaylock will not run without it.

### CODE REVIEW

File: src/session-lock.c [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+	struct wlr_session_lock_surface_v1 *lock_surface = data;
+	struct output *output = lock_surface->output->data;
+	struct session_lock_output *lock_output;
+	wl_list_for_each(lock_output, &lock->session_lock_outputs, link) {
+		if (lock_output->output == output) {
+			goto found_lock_output;
+		}
+	}
+	wlr_log(WLR_ERROR, "fatal: new lock surface, but no output");
+	exit(EXIT_FAILURE);
```
Comment (Consolatis) [inline src/session-lock.c:102]: Just had a quick look through the diff, this is the only place I saw a potential issue.

On some systems, turning off the monitors will cause them to actually be disconnected e.g. we don't have any outputs left. Could this trigger a complete exit of labwc when being locked and then turning off the monitors?

Comment (johanmalm) [reply src/session-lock.c]: Good point. Have changed to just returning.

**Comment by johanmalm** [comment]
Background info: https://github.com/swaywm/swaylock/commit/ac3b49b6571ceda3f8db11a98bfe320106996280

File: src/session-lock.c (hunk 1 above)
Comment (Consolatis) [reply src/session-lock.c]: Hm.. now I wonder if that could cause a lockscreen to not render a surface so there is no way to unlock the session once the output is back though. I guess some tests regarding that would be useful.

Comment (johanmalm) [reply src/session-lock.c]: It works fine.

The surface is created when the output comes back (in `session_lock_output_create()`).

I tested with the following:

```bash
#!/bin/sh

printf '%b\n' "starting..."
sleep 1

printf '%b\n' "eDP-1 off"
wlr-randr --output eDP-1 --off

sleep 2

printf '%b\n' "swaylock before"
./build/swaylock -c 222222 -d >$HOME/log.txt 2>&1 & 
printf '%b\n' "swaylock after"

sleep 2

printf '%b\n' "eDP-1 on"
wlr-randr --output eDP-1 --on

sleep 1
printf '%b\n' "complete"
```

Comment (Consolatis) [reply src/session-lock.c]: Thanks for confirming.

**Comment by Consolatis** [comment]
LGTM, didn't test.

**Comment by johanmalm** [comment]
Thanks for review.

Have just discovered a problem on multi-output setup. Need to investigate, so have put back to draft.

**Comment by johanmalm** [comment]
Helps if we set the node position :smile:

Ready now from my perspective. Will go for merge shortly.

**Pull request merged**